syntax = "proto3";

import "google/protobuf/empty.proto";
import "google/protobuf/timestamp.proto";

package files;

service Files {
  rpc UploadFile(UploadFileRequest) returns (FileURLResponse);
  rpc FileInfo(FileOperationRequest) returns (FileInfoResponse);
  rpc FileList(UserId) returns (FileListResponse);
  rpc DownloadFile(FileOperationRequest) returns (FileURLResponse);
  rpc DeleteFiles(DeleteFilesRequest) returns (google.protobuf.Empty);
  rpc DeleteAllFiles(UserId) returns (google.protobuf.Empty);
}

message DeleteFilesRequest {
  string user_id = 1;
  repeated string file_ids = 2;
}

message FileInfoResponse {
  string file_id = 1;
  string name = 2;
  string path = 3;
  uint64 size = 4;
  google.protobuf.Timestamp uploaded = 5;
}

message FileListResponse {
  repeated FileInfoResponse files = 1;
}

message FileOperationRequest {
  string user_id = 1;
  string file_id = 2;
}

message FileURLResponse {
  string url = 1;
}

message UploadFileRequest {
  string user_id = 1;
  string name = 2;
  string path = 3;
  uint64 size = 4;
}

message UserId {
  string user_id = 1;
}
